src/App.tsx

```diff
@@ -1,14 +1,53 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [loadOn, setLoadOn] = useState(false);
+  const [todos, setToDos] = useState<Todo[]>([]);
+  const [modalOn, setmodalOn] = useState(false);
+  const [modalToDo, setmodalTodo] = useState<Todo | null>(null);
+  const [sortType, setSortType] = useState<string>('');
+  const [query, setQuery] = useState<string>('');
+
+  useEffect(() => {
+    setLoadOn(true);
+    getTodos().then(todoos => setToDos(todoos)).finally(() => setLoadOn(false));
+  }, []);
+
+  function preparedTodos() {
+    let displayTodos = [...todos];
+
+    if (query) {
+      displayTodos = displayTodos.filter(
+        todo => todo.title.toLowerCase()
+          .includes(query.toLowerCase().trim()),
+      );
+    }
+
+    switch (sortType) {
+      case 'active':
+        return displayTodos.filter(todo => !todo.completed);
+
+      case 'completed':
+        return displayTodos.filter(todo => todo.completed);
+
+      case 'all':
+      default:
+        break;
+    }
```

Comment:
you dont need to add an "all" case, it has the same logic as the default, and you don't even need default, because It doesn't do anything

Reply:
Default case is always required for error handling. But yes, you can combine it with "All" case